content/en/docs/setup/install/multicluster/index.md

```diff
@@ -0,0 +1,281 @@
+    $ cat > base.yaml << EOF
+    apiVersion: install.istio.io/v1alpha2
+    kind: IstioControlPlane
+    spec:
+      values:
+        security:
+          selfSigned: false
+        global:
+          mtls:
+            enabled: true
+    EOF
+    {{< /text >}}
+
+1. Now, create the `prepcluster.sh` script. It'll be used to create the CA certificates and keys in all the clusters,
+and create common service accounts.
+
+    {{< text bash >}}
+    $ cat > prepca.sh << EOF
+    $ #!/bin/bash
+    $ pushd mesh-certs
+    $ make intermediate-$(kubectl config current-context)-certs
+    $ cd intermediate-$(kubectl config current-context)
+    $ kubectl create namespace istio-system >/dev/null || true
+    $ kubectl create secret generic cacerts -n istio-system \
+            --from-file=ca-cert.pem \
+            --from-file=ca-key.pem \
+            --from-file=root-cert.pem \
+            --from-file=cert-chain.pem \
+            --dry-run -o yaml| kubectl apply -f -
+    $ popd
+    $ kubectl create clusterrole istio-reader-istio-system --verb=get,list,watch \
+        --resource=nodes,pods,endpoints,services,replicasets.apps,replicationcontrollers --dry-run -o yaml \
+        | kubectl apply -f -
+     
+    $ kubectl create serviceaccount istio-reader-service-account -n istio-system  --dry-run -o yaml \
+        | kubectl apply -f -
+     
+    $ kubectl create clusterrolebinding istio-reader-istio-system --clusterrole=istio-reader-istio-system \
+         --serviceaccount=istio-system:istio-reader-service-account  --dry-run -o yaml \
+        | kubectl apply -f -
+    EOF
+    $ chmod +x prepca.sh
+    $ istioctl x multicluster generate -f mesh-topology.yaml --from base.yaml > istio-$(kubectl config current-context).yaml
+    $ istioctl manifest apply -f istio-$(kubectl config current-context).yaml
+    $ for CONTEXT in $(kubectl config get-context -o name); do
+        $ istioctl --context ${CONTEXT} x multicluster generate --from base.yaml -f mesh-topology.yaml > istio-${CONTEXT}.yaml
+        $ istioctl --context ${CONTEXT} manifest apply -f istio-${CONTEXT}.yaml
+    $ done
+    {{< /text >}}
+
+1. Now create an offline root CA by running the following:
+
+    {{< text bash >}}
+    $ mkdir mesh-certs && pushd mesh-certs
+    $ curl -O @samples/certs/Makefile@
+    $ export ROOTCA_ORG=mesh.example.com
+    $ export CITADEL_ORG=mesh.example.com
+    $ make root-ca
+    $ popd
+    {{< /text >}}
+
+    TBD: What should we recommend for thr ROOTCA_ORG and CITADEL_ORG values?
+    TBD: What advice can we give about keeping this secret?
+
+1. Finally, create a topology file. This will be used to hold the set of clusters you'll be adding to the mesh:
+
+    {{< text bash >}}
+    $ cat > mesh-topology.yaml << EOF
+    mesh_id: MyMeshID 
+    clusters:
+    EOF
+    {{< /text >}}
+    
+    You should set the mesh_id field to a reasonably unique and descriptive name for your mesh. This will help you identity your meshes in the
+    future.
+
```

Comment:
There are a few places with for loops over all contexts. It would be good to add a prereq step to verify only clusters-{1,2,3} are in the currently selected kubeconfig file.

Reply:
I don't see how that's possible. The premise here is that in fact, I've asked the user to create a custom kubeconfig file which only contains the clusters they care about. So it's a requirement on the user to not throw random stuff in there.

I'll make that more clear in the before you begin section.